include/seqan3/search/search.hpp

```diff
@@ -165,6 +168,26 @@ inline auto search(queries_t && queries,
 }
 
 //!\cond DEV
+// Convert query sequence if it does not match the alphabet type of the index.
+//!\overload
+template <fm_index_specialisation index_t,
+          std::ranges::forward_range queries_t,
+          typename configuration_t = decltype(search_cfg::default_configuration)>
+    requires std::ranges::forward_range<std::ranges::range_reference_t<queries_t>> &&
+             (!std::same_as<range_innermost_value_t<queries_t>, typename index_t::alphabet_type>)
```

Comment:
not the same but `convertible to` should be required?

Reply:
No, it must not be the same, because that means we need to convert it such that it is the same.
There's a static assert for the convertible thingie. but it is not needed for the overload resultion.

Reply:
ah ok :+1: